Test/SetVariableFromXml.Tests.ps1

```diff
@@ -0,0 +1,232 @@
+    )
+
+    $Content | Set-Content -Path (Join-Path -Path $TestDrive.FullName -ChildPath 'whiskey.yml')
+}
+
+function Init
+{
+    $script:path = $null
+    $script:context = $null
+}
+
+function ThenNoErrors
+{
+    It ('should not write any errors') {
+        $Global:Error | Should -BeNullOrEmpty
+    }
+}
+
+function ThenVariable
+{
+    param(
+        $Name,
+        $Is
+    )
+
+    It ('should create variable') {
+        $context.Variables.ContainsKey($Name) | Should -Be $true
+        $context.Variables[$Name] -join ',' | Should -Be ($Is -join ',')
+    }
+}
+
+function WhenRunningTask
+{
+    $Global:Error.Clear()
+
+    [Whiskey.Context]$context = New-WhiskeyTestContext -ForDeveloper -ConfigurationPath (Join-Path -Path $TestDrive.FullName -ChildPath 'whiskey.yml')
+    $parameter = $context.Configuration['Build'] | Where-Object { $_.ContainsKey('SetVariableFromXml') } | ForEach-Object { $_['SetVariableFromXml'] }
+    Invoke-WhiskeyTask -TaskContext $context -Name 'SetVariableFromXml' -Parameter $parameter
+    $script:context = $context
+}
+
+Describe 'SetVariableFromXml.when reading single element' {
+    Init
+    GivenXmlFile 'fubar.xml' @'
+<root>
+    <element1>one</element1>
+    <element2>two</element2>
+</root>
+'@
+    GivenWhiskeyYml @'
+Build:
+- SetVariableFromXml:
+    Path: fubar.xml
+    Variables:
+        Var1: /root/element1
+'@
+    WhenRunningTask
+    ThenVariable 'Var1' -Is 'one'
+}
+
+Describe 'SetVariableFromXml.when reading multiple elements' {
+    Init
+    GivenXmlFile 'fubar.xml' @'
+<root>
+    <element>one</element>
+    <element>two</element>
+</root>
+'@
+    GivenWhiskeyYml @'
+Build:
+- SetVariableFromXml:
+    Path: fubar.xml
+    Variables:
+        Var1: /root/element
+'@
+    WhenRunningTask
+    ThenVariable 'Var1' -Is @('one','two')
+}
+
+Describe 'SetVariableFromXml.when reading attribute' {
+    Init
+    GivenXmlFile 'fubar.xml' @'
+<root>
+    <element1 attr="one" />
+    <element2 attr="two" />
+</root>
+'@
+    GivenWhiskeyYml @'
+Build:
+- SetVariableFromXml:
+    Path: fubar.xml
+    Variables:
+        Var1: /root/element1/@attr
+'@
+    WhenRunningTask
+    ThenVariable 'Var1' -Is 'one'
+}
+
+Describe 'SetVariableFromXml.when reading multiple attributes' {
+    Init
+    GivenXmlFile 'fubar.xml' @'
+<root>
+    <element attr="one" />
+    <element attr="two" />
+</root>
+'@
+    GivenWhiskeyYml @'
+Build:
+- SetVariableFromXml:
+    Path: fubar.xml
+    Variables:
+        Var1: /root/element/@attr
+'@
+    WhenRunningTask
+    ThenVariable 'Var1' -Is @('one','two')
+}
+
+Describe 'SetVariableFromXml.when no element' {
+    Init
+    GivenXmlFile 'fubar.xml' @'
+<root>
+    <element attr="one" />
+</root>
+'@
+    GivenWhiskeyYml @'
+Build:
+- SetVariableFromXml:
+    Path: fubar.xml
+    Variables:
+        Var1: /root/element/fubar
+'@
+    WhenRunningTask
+    ThenVariable 'Var1' -Is ''
+    ThenNoErrors
+}
+
+Describe 'SetVariableFromXml.when XML has namespaces' {

```

Comment:
Maybe add a test for **multiple** namespaces other than just a default one?

Reply:
Added.